### ISSUE TITLE
After leaving on for a day or so, LT window goes white, unresponsive, have to force quit

### ISSUE THREAD

**Issue by snakecharmer1024** [LightTable/LightTable#2102]
I love LightTable, but I'll be switching back to Sublime Text until this is fixed. 

I'm using it on Mac OSX 10.10.5

Happens consistently, but only after about a day. I force quit, restart it and it works fine. Is troublesome when it does it while I'm working because I lose all my work since last save.

**Comment by cldwalker** [comment]
Sorry to hear. Handling an unresponsive event is on my short list which would make the force quit less painful. I'll leave this open to look into this. The losing work sucks and will be covered by #1700

**Comment by cldwalker** [comment]
Looked into handling unresponsive event by adding the following before [this line](https://github.com/LightTable/LightTable/blob/eaab2c10f579ebcc7e8af2707f0dd575bca82446/deploy/core/main.js#L65):

```js
  window.on("unresponsive", function() {
    console.log("PWNED!");
    alert("HANG!");
    window.destroy();
  });
```
Didn't see it handled. Pull requests welcome until we can get to this

**Comment by kenny-evitt** [comment]
@snakecharmer1024 I was curious if you are (or were) encountering the same thing I just ran into, but I just noticed the title change. My Light Table window is also (visibly) unresponsive and the entire window is blank (and gray):

![screen shot 2016-01-10 at 2 02 32 pm](https://cloud.githubusercontent.com/assets/1461730/12223357/dde8c0be-b7a2-11e5-8832-a5a7b6b8790a.png)

Activity Monitor reports the CPU Time of the process as 16:27. In 'realtime', I've had it open for at least a few days, tho I haven't been using it, or this computer, extensively. This is the first time I've run into this problem.

Activity Monitor also reports a "Virtual Memory Size" of 3.6 GB for the LT app process, which seems like it might be rather high. The "Electron Helper" process has a 2.5 GB virtual memory size.

I'm running OS X 10.11.2. I'm running a source build a few commits behind 0.8.0 (I think), but I don't expect any more-recent commits significantly affect this.

My log file is empty.

**Comment by kenny-evitt** [comment]
It seems this is an open issue upstream with Electron – see atom/electron#2409.

[I commented on that issue](https://github.com/atom/electron/issues/2409#issuecomment-170385197) and linked to a Gist with my crash dump file contents. It looks like an out-of-memory issue, tho I can't tell if it's Node that's the culprit or if it's just handling the out-of-memory condition.

**Comment by kenny-evitt** [comment]
@snakecharmer1024 If you would, please upload a crash dump (e.g. as a GitHub Gist or something similar) and link to it in a comment here and in atom/electron#2409. If you don't want to get bitten again by losing unsaved work, maybe it'd be sufficient just to open LT and leave it running, without actually using it.

**Comment by snakecharmer1024** [comment]
@kenny-evitt Yes that is exactly what I saw. I'll leave it on and post the crash dump. Where do I find the crash dump, again?

**Comment by kenny-evitt** [comment]
You might need to open the directory by running `open .` in *Terminal*, but I found mine in *~/Library/DiagnosticReport* (I think). It's the Mac OS X standard location for crash dumps so Google should confirm if my memory is accurate.

That said, an Electron person replied to me on their issue and stated that LT is to blame in my case and that we should track down the likely memory leak on our end. So we should be running with Dev Inspector open and periodically checking memory usage (if that's not too much of a pain).
